### PR TITLE
RavenDB-11906

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexOperationBase.cs [hunk 1]
```diff
@@ -37,39 +35,43 @@ protected IndexOperationBase(Index index, Logger logger)
             _logger = logger;
         }
 
-        protected static RavenPerFieldAnalyzerWrapper CreateAnalyzer(Func<Analyzer> createDefaultAnalyzer, IndexDefinitionBase indexDefinition, bool forQuerying = false)
+        protected static RavenPerFieldAnalyzerWrapper CreateAnalyzer(Func<LowerCaseKeywordAnalyzer> createDefaultAnalyzer, IndexDefinitionBase indexDefinition, bool forQuerying = false)
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexOperationBase.cs]: Why this change?

File: src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs [hunk 2]
```diff
@@ -108,37 +108,57 @@ public dynamic LoadDocument(object keyOrEnumerable, string collectionName)
...
 
+        protected IEnumerable<AbstractField> CreateField(string name, object value, CreateFieldOptions options)
+        {
+            // IMPORTANT: Do not delete this method, it is used by the indexes code when using CreateField
+
+            options = options ?? CreateFieldOptions.Default;
+
+            IndexFieldOptions allFields = null;
+            var scope = CurrentIndexingScope.Current;
+            if (scope.IndexDefinition is MapIndexDefinition mapIndexDefinition)
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs:119]: what about map /reduce indexes?

Comment (ayende) [reply src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs:119]: JS indexes?

File: src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexOperationBase.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexOperationBase.cs]: Just to be clear what default analyzer actually is. All places that call `CreateAnalyzer` are creating instance of `LowerCaseKeywordAnalyzer`

File: src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs (hunk 2 above)
Comment (ppekrol) [reply src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs:119]: MapReduceIndexDefinition inherits from MapIndexDefinition so it is fine.

JS indexes do not support 'CreateField' at all

File: src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexOperationBase.cs (hunk 1 above)
Comment (ayende) [reply src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexOperationBase.cs]: But we may want to change that in the future, and I don't see what benefit the lowered type gives us here?

File: src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs (hunk 2 above)
Comment (ayende) [reply src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs:119]: You can certainly do dynamic fields in JS, see:

```
 @"map('Users', function (u){
                                    return {
                                        Name: u.Name,
                                        _: {$value: u.Name, $name:'AnalyzedName', $options:{index: true, store: true}}
                                    };
                                })",
```

File: src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexOperationBase.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexOperationBase.cs]: fixed

File: src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs (hunk 2 above)
Comment (ppekrol) [reply src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs:119]: I will check that tomorrow

Comment (ppekrol) [reply src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs:119]: Done. Changed syntax a bit to cover more options.

Comment (ppekrol) [reply src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs:119]: 'storage' supports both:
- true/false
- 'Yes' / 'No'

File: test/FastTests/Client/Indexing/JavaScriptIndexTests.cs [hunk 3]
```diff
@@ -665,7 +665,7 @@ public UsersByNameAndAnalyzedName()
                             @"map('Users', function (u){
                                     return {
                                         Name: u.Name,
-                                        _: {$value: u.Name, $name:'AnalyzedName', $options:{index: true, store: true}}
+                                        _: {$value: u.Name, $name:'AnalyzedName', $options:{indexing: 'Search', storage: true}}
```
Comment (ayende) [inline test/FastTests/Client/Indexing/JavaScriptIndexTests.cs:668]: isn't this a breaking change?
This is experimental feature, though.

Comment (ppekrol) [reply test/FastTests/Client/Indexing/JavaScriptIndexTests.cs:668]: This was my thinking. We can support both if needed but I do not think that this is a good idea in this case. Can you make a call here?

Comment (ayende) [reply test/FastTests/Client/Indexing/JavaScriptIndexTests.cs:668]: It's fine to make this change, yes

**Review by arekpalinski — APPROVED** [review]
